### PR TITLE
Implement GetArtifactSummary

### PR DESCRIPTION
**Pull request by tenstad**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

Closes https://github.com/jfrog/jfrog-client-go/issues/528

### CODE REVIEW

**Comment by tenstad** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by eyalbe4** [comment]
recheckcla

File: xray/services/summary.go [hunk 1]
```diff
@@ -53,6 +55,59 @@ func (ss *SummaryService) GetBuildSummary(params XrayBuildParams) (*SummaryRespo
...
+type ArtifactSummaryResponse struct {
+	Artifacts []Artifact `json:"artifacts,omitempty"`
+	Errors    []Error    `json:"errors,omitempty"`
+}
+
+type Artifact struct {
+	General General `json:"general,omitempty"`
+	Issues  []Issue `json:"issues,omitempty"`
+	// TODO: Create License struct with correct fields for api/v2/summary/artifact endpoint
+	// Licenses []License `json:"licenses,omitempty"`
```
Comment (tenstad) [inline xray/services/summary.go:100]: A `License` struct already exists https://github.com/jfrog/jfrog-client-go/blob/e3c6206b724288677c125d89268f461a78c33e57/xray/services/scan.go#L225-L231, but does not match the api specifiaction of https://www.jfrog.com/confluence/display/XRAY2X/Xray+REST+API#XrayRESTAPI-ArtifactSummary .

Is the existing one correct, and should I simply create a new `ArtifactLicense` struct?

**Comment by eyalbe4** [comment]
Thanks for this contribution @tenstad!
Will you be able to add the new API to the README?

**Comment by tenstad** [comment]
> Thanks for this contribution @tenstad! Will you be able to add the new API to the README?

Ofc! Added a new section to the README now 😊

We could skip the first two lines where `summaryService` is created. I included them because `GetArtifactSummary` requires a `SummaryService` with `XrayDetails`, which I do not find intuitive to create. Maybe I am doing something wrong.

**Comment by tenstad** [comment]
> which I do not find intuitive to create. Maybe I am doing something wrong.

Ah, it is because I did not implement a wrapper method on `XrayServiceManager`. I missed this because `GetBuildSummary` does not have such a method.

Fixed for `GetArtifactSummary` now, and I suggest that `GetBuildSummary` is added to the README and a corresponding wrapper is created for it in another PR.

File: xray/services/summary.go (hunk 1 above)
Comment (mikaello) [reply xray/services/summary.go:100]: The existing [`License` struct](https://github.com/jfrog/jfrog-client-go/blob/e3c6206b724288677c125d89268f461a78c33e57/xray/services/scan.go#L225-L231) belongs to the `/scan/graph` API it seems, when comparing to the TypeScript library and Java library they have created two different datatypes for the license used in scan graph summary and in summary:

*TypeScript*:
* [Scan/GraphLicense.ts](https://github.com/jfrog/jfrog-client-js/blob/master/model/Xray/Scan/GraphLicense.ts)
* [Summary/License.ts](https://github.com/jfrog/jfrog-client-js/blob/master/model/Xray/Summary/License.ts)

*Java*:
* [scan/License.java](https://github.com/jfrog/xray-client-java/blob/master/src/main/java/com/jfrog/xray/client/services/scan/License.java)
* [summary/License.java](https://github.com/jfrog/xray-client-java/blob/master/src/main/java/com/jfrog/xray/client/services/summary/License.java)

So I think it is safe to say that you should create a new license struct.

File: xray/services/summary.go [hunk 2]
```diff
@@ -53,6 +55,59 @@ func (ss *SummaryService) GetBuildSummary(params XrayBuildParams) (*SummaryRespo
...
+}
+
+type General struct {
+	ComponentId string `json:"component_id,omitempty"`
+	Name        string `json:"name,omitempty"`
+	Path        string `json:"path,omitempty"`
+	PkgType     string `json:"pkg_type,omitempty"`
+	Sha256      string `json:"sha256,omitempty"`
+}
```
Comment (mikaello) [inline xray/services/summary.go:109]: When comparing to [TypeScript](https://github.com/jfrog/jfrog-client-js/blob/master/model/Xray/Summary/General.ts) and [Java](https://github.com/jfrog/xray-client-java/blob/master/src/main/java/com/jfrog/xray/client/services/summary/General.java), you are missing the following properties:

* `Sha1`
* `ParentSha256`

I notice that they are not part of [the documentation](https://www.jfrog.com/confluence/display/XRAY2X/Xray+REST+API#XrayRESTAPI-ArtifactSummary), but I guess the Go lib should document the same API as TypeScript and Java?!

File: xray/services/summary.go (hunk 1 above)
Comment (eyalbe4) [reply xray/services/summary.go:100]: Right - please go ahead and create it. We rally prefer not to merge PRs with // TODO comments or commented out code @mikaello .

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Let's also add a test for new ```GetArtifactSummary``` API. The test can performed using a mock server. You can look at the ```TestNewXrayScanService``` as a reference and do something similar.
You can put the test inside the ```summary_test.go``` file under the ```tests``` directory. You'll see all the test files there.

**Comment by tenstad** [comment]
@eyalbe4 I added a simple test now :) It can definitely be improved/extended in the future, but at least a basic response with all fields is now tested.

Had to add two fields that were missing in existing structs (ComponentPhysicalPaths and Cwe), so that the entire sample response from xray could be retured by the client.

**Pull request merged**